src/ProxyServer/ProxyServer.swift

```diff
@@ -29,6 +29,8 @@ open class ProxyServer: NSObject, TunnelDelegate {
     open var observer: Observer<ProxyServerEvent>?
 
     var tunnels: TunnelArray = []
+    
+    private let accessQueue = DispatchQueue(label: "NEKit.ProxyServer.TunnelArrayAccess", attributes: .concurrent)
```

Comment:
Strangely, I wrote some Logs and asserts that `tunnels` did run in a serial queue, but when I revert the commit `e19449e`, Xcode told me it leaked and I did not know where the reason was.

![image](https://cloud.githubusercontent.com/assets/548532/23785297/fdf1e360-05a1-11e7-8c57-e04823da1ae8.png)


Although it seems to leak, but running a few hours, the VPN memory usage is very stable.

PS:
  `accessQueue` is a concurrent queue, but `barrier_asyn` make writes synchronized.